### PR TITLE
fix the issue#5275: iso_create doesn't add folders

### LINKED ISSUE #5275: iso_create doesn't add folders
### Summary

When trying to add a folder/directory to an iso, using `/path/to/folder` in `src_files`,  the module fails. From the output I can see ansible is parsing the content of the folder, but it doesn't add it to the iso.

### Issue Type

Bug Report

### Component Name

iso_create

### Ansible Version

```
# RHEL VM
$ ansible --version
ansible [core 2.12.2]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/radam/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.8/site-packages/ansible
  ansible collection location = /home/radam/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.8.12 (default, Sep 16 2021, 10:46:05) [GCC 8.5.0 20210514 (Red Hat 8.5.0-3)]
  jinja version = 2.10.3
  libyaml = True

# Arch Linux VM
$ ansible --version
ansible [core 2.13.3]
  config file = None
  configured module search path = ['/home/radu/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/radu/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/radu/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/radu/.local/bin/ansible
  python version = 3.10.6 (main, Aug  3 2022, 17:39:45) [GCC 12.1.1 20220730]
  jinja version = 3.1.2
  libyaml = True
```

### Community.general Version

```console (paste below)
>>>  RHEL VM
$ ansible-galaxy collection list community.general

# /home/radam/.ansible/collections/ansible_collections
Collection        Version
----------------- -------
community.general 4.8.0

>>> Arch Linux VM
$ ansible-galaxy collection list community.general

# /home/radu/.ansible/collections/ansible_collections
Collection        Version
----------------- -------
community.general 5.5.0  
```

### Configuration

Folder structure used:
```
files
├── a_simple_text_file.txt
└── scripts
    ├── 1.txt
    ├── 2.txt
    └── 3.txt
```

### OS / Environment

I did tests on two VMs':
RHEL 8.6
Arch Linux

### Steps to Reproduce

Using create_cd.yml playbook as below:
```yaml
---
- name: Build CD
  hosts: localhost

  tasks:
    - name: Create iso
      community.general.iso_create:
        src_files:
          - "files/a_simple_text_file.txt"
          - "files/scripts"
        dest_iso: "mycd.iso"
        interchange_level: 3
        joliet: 3
        vol_ident: "test"
```

Running the playbook:
```
$ ansible-playbook create_cd.yml
```

### Expected Results

Expected to have an iso containing `a_simple_text_file.txt` and the entire `scripts` subfolder.

### Actual Results

```console (paste below)
$ ansible-playbook create_cd.yml
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [Build VM] **************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] *******************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [Create iso] ************************************************************************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Failed to add file files/scripts/1.txt to ISO file due to Could not find path"}

PLAY RECAP *******************************************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0  
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by ZouYuhua**
##### SUMMARY
Fixes #5275

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
iso_create

##### ADDITIONAL INFORMATION

**Steps:**
1. reproduce the issue in local machine
2. modify iso_create.py
3. run test in local machine
4. check the iso content manually

**Folder structure used:**
```
files
├── a_simple_text_file.txt
└── scripts
    ├── 1.txt
    ├── 2.txt
    └── 3.txt
```

**The content of test file:**
```
# Copyright 2021-2022 VMware, Inc.
# SPDX-License-Identifier: BSD-2-Clause
---
# Main playbook for launching guest OS validation testing on vSphere
- name: Build CD
  hosts: localhost
  tasks:
    - name: Create iso
      community.general.iso_create:
        src_files:
          - "files/a_simple_text_file.txt"
          - "files/scripts"
        dest_iso: "mycd.iso"
        interchange_level: 3
        joliet: 3
        vol_ident: "test"

    - name: Create iso with UDF 
      community.general.iso_create:
        src_files:
          - "files/a_simple_text_file.txt"
          - "files/scripts"
        dest_iso: "mycd_udf.iso"
        interchange_level: 3
        joliet: 3
        vol_ident: "test_udf"
```

**For step 1:  I can reproduce the issue**
root [ /mnt ]# ansible-playbook test_create_iso.yml 
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [Build CD] *********************************************************************************************************************************************

TASK [Gathering Facts] **************************************************************************************************************************************
ok: [localhost]

TASK [Create iso] *******************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Failed to add file files/scripts/1.txt to ISO file due to Could not find path"}

**For step 3:  run test in local machine: succed to create iso**
root [ /mnt ]# ansible-playbook test_create_iso.yml 
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [Build CD] *********************************************************************************************************************************************

TASK [Gathering Facts] **************************************************************************************************************************************
ok: [localhost]

TASK [Create iso] *******************************************************************************************************************************************
changed: [localhost]

TASK [Create iso with UDF] **********************************************************************************************************************************
changed: [localhost]

PLAY RECAP **************************************************************************************************************************************************
localhost                  : ok=3    changed=2    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0 

**For step 4: mount the created mycd.iso and mycd_pdf.iso and check the content**

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Can you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -179,6 +179,7 @@ Bugfixes
 - filesystem - improve error messages when output cannot be parsed by including newlines in escaped form (https://github.com/ansible-collections/community.general/pull/4700).
 - funcd connection plugin - fix signature of ``exec_command`` (https://github.com/ansible-collections/community.general/pull/5111).
 - ini_file - minor refactor fixing a python lint error (https://github.com/ansible-collections/community.general/pull/5307).
+- iso_create plugin - iso_create doesn't add folers (https://github.com/ansible-collections/community.general/pull/5468).
```
Comment (felixfontein) [inline CHANGELOG.rst:182]: Please remove this and the other changelog related changes, and instead create a file in `changelogs/fragments/` with the content you added to the YAML file as described [here](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment).

Comment (ZouYuhua) [reply CHANGELOG.rst]: Thanks. Done

File: changelogs/fragments/5468-iso-create-not-add-folders.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - iso_create - iso_create doesn't add folders (https://github.com/ansible-collections/community.general/issues/5275).
```
Comment (felixfontein) [inline changelogs/fragments/5468-iso-create-not-add-folders.yml:2]:
```suggestion
  - iso_create - the module somtimes failed to add folders for Joliet and UDF formats (https://github.com/ansible-collections/community.general/issues/5275).
```

**Comment by felixfontein** [comment]
@ZouYuhua thanks for fixing this!

**Pull request merged**